### PR TITLE
#187300192 User Login authentication

### PR DESCRIPTION
**Pull request by kayigmb**
#### What does this PR do?
- User login for the buyer, admin and seller
#### Description of Task to be completed?
 - The user is able to login and the crediential are stored for future use
#### How should this be manually tested?
 - using npm run dev
#### Any background context you want to provide?
 - The user is able to login

#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/story/show/187300192
#### Screenshots (if appropriate)
image
#### Questions:

### CODE REVIEW

**Review by tchamianest — APPROVED** [review]

**Review by emmanueltct — APPROVED** [review]

File: src/app/utils/LoginValidation.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import z from "zod";
```
Comment (Eli250) [inline src/app/utils/LoginValidation.tsx:1]: This file should be under `src/validations` directory. `app` directory is only meant for pages and api's in Next14.

File: src/app/pages/auth/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
+"use client";
```
Comment (Eli250) [inline src/app/pages/auth/login/page.tsx:1]: File naming pattern is wrong. Get rid of pages directory as it is useless. The pattern should be `src/app/auth/login/page.tsx`

File: src/app/pages/auth/login/otpVerify.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
+"use client"
```
Comment (Eli250) [inline src/app/pages/auth/login/otpVerify.tsx:1]: If the content of these codes are just a compent for a certain page, move them under components directory. And also don't use pages anywhere else. You can move this file into `scr/components/VerifyOTP` for example.

**Review by Eli250 — CHANGES_REQUESTED** [review]
Don't use pages under app directory. Apply requested changes.

File: src/app/auth/login/page.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+import { useRef, useState } from "react";
+import Link from "next/link";
+import Button from '@/components/Button';
+import InputBox from "@/components/InputBox";
+import Image from "next/image"
+import loginValidation from "@/validations/LoginValidation";
+import OtpVerify from "@/components/otpVerify";
+
+export default function Login() {
+    const [email, setEmail] = useState('');
```
Comment (Eli250) [inline src/app/auth/login/page.tsx:14]: Separate component logic from UI components. Use custom hooks as we have seen during capstone project.

**Review by Eli250 — CHANGES_REQUESTED** [review]

**Review by Eli250 — APPROVED** [review]

**Review by Eli250 — APPROVED** [review]

**Pull request merged**